storage/s3_storage.py

```diff
@@ -57,6 +63,10 @@ def save_to_directory(self, directory_path: str) -> None:
         client = self._connect()
         directory_prefix = "{}/".format(self._keyname)
         dir_object = client.list_objects(Bucket=self._bucket, Prefix=directory_prefix)
+
+        if "Contents" not in dir_object:
+            raise NotFoundError("No Files Found")
```

Comment:
Similar to my comment in the Google implementation, we need to handle the case where the `download_file` call on line 82 fails because the file doesn't exist in the S3 bucket, anymore.

The same goes for the `delete_directory` function.